### ISSUE TITLE
Inclusion of `vite-plugin-glob` conflicts with core vite >= 3 functionality

### ISSUE THREAD

**Issue by mcous** [brillout/vite-plugin-ssr#431]
### Description

I'm trying to use `import.meta.glob` to get a list of all child pages as described in https://github.com/brillout/vite-plugin-ssr/issues/230#issuecomment-1102494751. However, I'm running into an issue that seems like it's caused by the fact that `vite-plugin-ssr` pulls in the now-deprecated [`vite-plugin-glob`](https://github.com/antfu/vite-plugin-glob).

I'm trying to import the `title` export of  my `mdx` pages, with tree shaking, as per the [`vite` documentation for `import.meta.glob`](https://vitejs.dev/guide/features.html#named-imports) and the suggestion in the issue thread above:

```ts
// pages/blog/index.page.ts
// ...
export async function onBeforeRender(
  pageContext: PageContextClient
): Promise<{ pageContext: Partial<PageContext> }> {
  const posts = import.meta.glob('./*.page.mdx', { import: 'title', eager: true })
  console.log('Posts', posts)
  return { pageContext: {} }
}
```

This conflicts with the [same documentation for [`vite-plugin-glob`](https://github.com/antfu/vite-plugin-glob#named-imports), which uses the option name `export` instead:

```ts
// pages/blog/index.page.ts
// ...
export async function onBeforeRender(
  pageContext: PageContextClient
): Promise<{ pageContext: Partial<PageContext> }> {
  const posts = import.meta.glob('./*.page.mdx', { export: 'title', eager: true })
  console.log('Posts', posts)
  return { pageContext: {} }
}
```

However, argument checking in both vite core and `vite-plugin-glob` prevent be from using either. I have been unable to find a workaround for this issue so far, other than importing the entire child page module.

### Error Stack

If I use the option name `import` as in the vite docs, I get an error raised from `vite-plugin-glob`:

```shell
Error: Invalid glob import syntax: Unknown options import
    at err (/Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite-plugin-glob@0.2.8/node_modules/vite-plugin-glob/dist/index.cjs:91:17)
    at /Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite-plugin-glob@0.2.8/node_modules/vite-plugin-glob/dist/index.cjs:159:17
    at Array.map (<anonymous>)
    at parseImportGlob (/Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite-plugin-glob@0.2.8/node_modules/vite-plugin-glob/dist/index.cjs:87:24)
    at transform (/Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite-plugin-glob@0.2.8/node_modules/vite-plugin-glob/dist/index.cjs:202:23)
    at TransformContext.transform (/Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite-plugin-glob@0.2.8/node_modules/vite-plugin-glob/dist/index.cjs:66:28)
    at Object.transform (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:35579:53)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async loadAndTransform (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:39888:29)
```

However, if I use the option name `export` as spec'd by `vite-plugin-glob`, I get an error from vite core, instead:

```shell
Error: Invalid glob import syntax: Unknown options export
    at err (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:35679:23)
    at file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:35791:27
    at Array.map (<anonymous>)
    at parseImportGlob (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:35675:27)
    at transformGlobImport (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:35840:27)
    at TransformContext.transform (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:35646:34)
    at Object.transform (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:35579:53)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async loadAndTransform (file:///Users/mc/projects/mcous.github.io/node_modules/.pnpm/vite@3.0.9/node_modules/vite/dist/node/chunks/dep-0fc8e132.js:39888:29)
```

**Comment by brillout** [comment]
Vps installs `vite-plugin-glob@0.2.8`; check the documentation for that version instead: https://github.com/antfu/vite-plugin-glob/tree/v0.2.8.

Also vps doesn't set `vite-plugin-glob`'s option `takeover` to `true`. So, in principle, you should be able to use Vite's built-in implementation.

- `import.meta.importGlob()` => `vite-plugin-glob`
- `import.meta.glob()` => Vite

**Comment by mcous** [comment]
Documentation for the version included with VPS also uses `export` rather than `import`. I agree that it's weird that they'd interact, but I'm certainly getting the stack traces I posted, and I'm only ever using `import.meta.glob`...

I'll see if I can get up a minimal repro repository set up in a few hours and post my results here

**Comment by mcous** [comment]
@brillout I've confirmed that merely including `vite-plugin-glob` in any way (regardless of the `takeover` option) breaks the use of the `import` option for vanilla `import.meta.glob`. Reproduction available here: https://github.com/mcous/vps-431

Happily enough, I've also discovered a workaround if `vite-plugin-glob` is installed - simply use the plugin version with the `export` option: `import.meta.importGlob({ export: 'title' })`

**Comment by brillout** [comment]
Released `vite-plugin-ssr@0.4.29` which doesn't use `vite-plugin-glob` anymore.

You should now be able to use Vite's built-in `import.meta.glob()` feature.

FYI, in case you(r company) is interested: https://github.com/sponsors/brillout.
